tests/test_mdspan_ctors.cpp

```diff
@@ -45,20 +45,37 @@
 #include <vector>
 
 #include <gtest/gtest.h>
+#include "offload_utils.hpp"
 
 namespace stdex = std::experimental;
 _MDSPAN_INLINE_VARIABLE constexpr auto dyn = stdex::dynamic_extent;
 
+
+void test_mdspan_ctor_data_carray() {
+  size_t* result = allocate_array<size_t>(6);
+  dispatch([=] _MDSPAN_HOST_DEVICE () {
+    int data[1] = {42};
+    stdex::mdspan<int, stdex::extents<1>> m(data);
+    result[0] = m.data() == data ? 1 : 0;
```

Comment:
```C++
#define POOR_MAN_ASSERT_EQ(LHS, RHS) \
if (!(LHS == RHS)) {
  printf("expected equality of %s and %s", #LHS, #RHS); // can improve to print values when possible
  ++errors; // increment some error counter
}

Reply:
@dalg24 I have a vague memory that the Sierra (Sandia's finite-element suite) developers had written something like this.

Reply:
addressed